### PR TITLE
NEXT-12778 [NEW] Using base components

### PR DESCRIPTION
**Pull request by NiklasLimberg**
What should be done?

Create a new article on our GitBook instance which explains how to use base administration components.

This article should mention:

- The prerequisite, a working plugin (Refer to the plugin base guide)
- Every other prerequisite you figure out during writing the guide (e.g. a subscriber, knowing how to create a service, a controller, etc.)
- A short code example, including an explanation, on how to do it

Category: Extensions > Plugins > Administration
Are there already guides in the previous documentation for this?

Unfortunately, No. But please have a look at our example guide(e.g. note the PLACEHOLDERs for cross-references, that do not exist yet) and the Writing guidelines.

For more information, ask me, Patrick Stahl.

### CODE REVIEW

File: guides/plugins/plugins/administration/using-base-components.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Using base components
+
+The Shopware 6 Administration registers all components into the component registry wich makes them available in the entire Administration
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:3]:
```suggestion
The Shopware 6 Administration registers all components into the component registry wich makes them available in the entire Administration.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework, used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Finding the base component needed
+
+All Shopware 6 Administration components can be seen in the [Component Library](https://component-library.shopware.com/). 
+There you can see what each of the components does and looks like, it also shows you what props the can work with and wich slots they have.
+
+## Using the base
+
+As mentioned before in the introduction all components used in the Shopware 6 Administration are registered in the `component registry`, which makes them accessible everywhere.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:16]:
```suggestion
As mentioned before in the introduction, all components used in the Shopware 6 Administration are registered in the `component registry`, which makes them accessible everywhere.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Learn what the `component registry` is and how you can register your own components to it [here](./add-custom-component.md)
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:17]:
```suggestion
Learn about the `component registry` how you can register your own components to it [in the corresponding guide](./add-custom-component.md)
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+In this example we will use the `sw-text-field` in our template.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:19]:
```suggestion
Using base components in your own is rather simple. In the example below we will use the `sw-text-field` in our template:
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+{% code title="<plugin-root>/src/Resources/app/administration/app/src/component/example-component/example.html.twig" %}
+```html
+<div>
+    <sw-text-field />
+</div>
+```
+{% endcode %}
+
+Learn how to write templates and how to include them in your components [here](./writing-templates.md).
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:29]:
```suggestion
That's basically it. To continue building beautiful custom components, learn how to write templates and how to include them in your components [here](./writing-templates.md).
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+As mentioned before in the introduction all components used in the Shopware 6 Administration are registered in the `component registry`, which makes them accessible everywhere.
+Learn what the `component registry` is and how you can register your own components to it [here](./add-custom-component.md)
+
+In this example we will use the `sw-text-field` in our template.
+
+{% code title="<plugin-root>/src/Resources/app/administration/app/src/component/example-component/example.html.twig" %}
+```html
+<div>
+    <sw-text-field />
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:24]: This example and the whole paragraph feels rather short and abrupt. I already tries to improve that thorough using more text. Could you use an example closer to the real world?

**Review by leichteckig — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/administration/using-base-components.md (hunk 6 above)
Comment (NiklasLimberg) [reply guides/plugins/plugins/administration/using-base-components.md:24]: Well then it would proably overlap to much with the [writing templates](https://github.com/shopware/docs/blob/master/guides/plugins/plugins/administration/writing-templates.md) article, wich itself already overlaps with the [add-custom-component](https://github.com/shopware/docs/blob/master/guides/plugins/plugins/administration/add-custom-component.md) article

Comment (leichteckig) [reply guides/plugins/plugins/administration/using-base-components.md:24]: I know, that's tough 🙈  Maybe using a bigger example then? With some probs handled into `sw-text-field` or something? 🙈

**Review by leichteckig — APPROVED** [review]

File: guides/plugins/plugins/administration/using-base-components.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# Using base components
+
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:2]: You're starting very abrupt.
Please write an intro, like: The Shopware 6 Administration comes with a bunch of tailored Vue-components... or sth. like this

File: guides/plugins/plugins/administration/using-base-components.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework, used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Finding the base component needed
+
+All Shopware 6 Administration components can be seen in the [Component Library](https://component-library.shopware.com/). 
+There you can see what each of the components does and looks like, it also shows you what props the can work with and wich slots they have.
+
+## Using the base
+
+As mentioned before in the introduction, all components used in the Shopware 6 Administration are registered in the `component registry`, which makes them accessible everywhere.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:16]: You are directly contradicting to your sentence before, where you state, that the components are available anywhere in the Administration.

```suggestion
As mentioned before in the introduction, all components used in the Shopware 6 Administration are registered in the `component registry`, which makes them accessible in the Administration.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+Learn about the `component registry` how you can register your own components to it [in the corresponding guide](./add-custom-component.md)
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:17]: I'd put this way up and write it like this:

This guide shows you how to use Shopware's own pre-built components. If you want to learn, how to create your custom components or the component registry in general, s. Link to page.

File: guides/plugins/plugins/administration/using-base-components.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+```
+{% endcode %}
+
+That's basically it. To continue building beautiful custom components, learn how to write templates and how to include them in your components [here](./writing-templates.md).
+
+## Next steps
+
+Now that you know that all components are accessible through the component registry, you might want to learn more about the following things:
+
+* [How wirte templates](./writing-templates.md)
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:35]:
```suggestion
* [How to create custom templates](./writing-templates.md)
```

**Review by lernhart — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/administration/using-base-components.md (hunk 9 above)
Comment (NiklasLimberg) [reply guides/plugins/plugins/administration/using-base-components.md]: i've rewritten the introducton. i hope you are happy with it

Comment (lernhart) [reply guides/plugins/plugins/administration/using-base-components.md]: Perfect :+1:

File: guides/plugins/plugins/administration/using-base-components.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
+# Using base components
+
+The Shopware 6 Administration comes with a bunch of tailored Vue-components, already accessible in all of your templates trough to `component registry`.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:3]:
```suggestion
The Shopware 6 Administration comes with a bunch of tailored Vue-components, already accessible in all of your templates via the `component registry`.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+This guide will show you how you can use Shopware made Components in your templates,
```
Comment (lernhart) [inline guides/plugins/plugins/administration/using-base-components.md:4]:
```suggestion
This guide will show you how you can use Shopware-made components in your templates, 
```

**Review by lernhart — APPROVED** [review]

**Comment by lernhart** [comment]
@NiklasLimberg i accidentally approved your changes, please implement my requests nevertheless :D

File: guides/plugins/plugins/administration/using-base-components.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+if you want to learn more about the `component registry` and how you can register your own components to it have a look at the [corresponding guide](./add-custom-component.md)
+
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework, used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Finding the base component needed
+
+All Shopware 6 Administration components can be seen in the [Component Library](https://component-library.shopware.com/). 
+There you can see what each of the components does and looks like, it also shows you what props the can work with and wich slots they have.
```
Comment (PaddyS) [inline guides/plugins/plugins/administration/using-base-components.md:14]:
```suggestion
There you can see what each of the components does and looks like, it also shows you what props they can work with and which slots they have.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    <sw-text-field />
+</div>
+```
+{% endcode %}
+
+That's basically it. To continue building beautiful custom components, learn how to write templates and how to include them in your components [here](./writing-templates.md).
+
+## Next steps
+
+Now that you know that all components are accessible through the component registry, you might want to learn more about the following things:
```
Comment (PaddyS) [inline guides/plugins/plugins/administration/using-base-components.md:34]: What really confuses me here:
You're talking a lot about the component registry and "things being accessible through the component registry", yet all that's done is "simply using" the component in the templates. This way it feels like there's missing something, some kind of "Component.Registry.getElement". I know this is not a thing, but the guide kinda makes me feel like such a thing is missing.

TL;DR: Maybe at least link the "component registry", if it's a class or something?

**Review by PaddyS — CHANGES_REQUESTED** [review]

**Comment by PaddyS** [comment]
Also, rebase & squash please

File: guides/plugins/plugins/administration/using-base-components.md (hunk 14 above)
Comment (NiklasLimberg) [reply guides/plugins/plugins/administration/using-base-components.md:34]: The `Component.Registry.getElement `thing is approximately what internally happens, when a component is used in a template. i'll write a sentence about it

Comment (NiklasLimberg) [reply guides/plugins/plugins/administration/using-base-components.md:37]: i hope its clearer now

**Review by lernhart — CHANGES_REQUESTED** [review]
I'm okay with the article.

However, please rebase and squash your commits into one.

File: guides/plugins/plugins/administration/using-base-components.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
+# Using base components
+
+The Shopware 6 Administration comes with a bunch of tailored Vue-components, already accessible in all of your templates via the `component registry`.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:3]:
```suggestion
The Shopware 6 Administration comes with a bunch of tailored Vue components, already accessible in all of your templates via the `component registry`.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+This guide will show you how you can use Shopware-made components in your templates, 
+if you want to learn more about the `component registry` and how you can register your own components to it have a look at the [corresponding guide](./add-custom-component.md)
+
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework, used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:9]:
```suggestion
All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+## Finding the base component needed
+
+All Shopware 6 Administration components can be seen in the [Component Library](https://component-library.shopware.com/).
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:13]:
```suggestion
All Shopware 6 Administration components can be found in the [Component Library](https://component-library.shopware.com/). 
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with [Vue](https://vuejs.org/), the framework, used in the Administration. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Finding the base component needed
+
+All Shopware 6 Administration components can be seen in the [Component Library](https://component-library.shopware.com/). 
+There you can see what each of the components does and looks like, it also shows you what props they can work with and which slots they have.
+
+## Using the base
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:16]:
```suggestion
## Using the base component
```

File: guides/plugins/plugins/administration/using-base-components.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+As mentioned before in the introduction, all components used in the Shopware 6 Administration are first registered to the `component registry`.
+Wich `component registry` is just a map of all components, wich then get registered to Vue during the `Administrations boot process`.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/using-base-components.md:19]:
```suggestion
This `component registry` is just a map of all components, which then get registered to Vue during the `Administrations boot process`.
```

**Review by leichteckig — CHANGES_REQUESTED** [review]

**Review by lernhart — APPROVED** [review]

**Review by leichteckig — APPROVED** [review]
